app/controllers/vendors/records_controller.rb

```diff
@@ -61,23 +74,47 @@ def assemble_alert(failed_files)
     end
 
     # Take xml file name, data, error collection hash, return CQM::Patient
-    def add_patient(name, data, failed_files)
+    def add_patient(name, data, failed_files, validator)
       doc = Nokogiri::XML::Document.parse(data)
       doc.root.add_namespace_definition('cda', 'urn:hl7-org:v3')
       doc.root.add_namespace_definition('sdtc', 'urn:hl7-org:sdtc')
 
       # basic CDA schema validation
-      validator = CDA.instance
       errors = validator.validate(doc)
       unless errors.empty?
         failed_files[name] = errors.map(&:message)
-        return nil
+        return nil, 0
+      end
+
+      time_shifted_patient(doc)
+    end
+
+    def time_shifted_patient(doc)
+      # check for start date
+      year_validator = MeasurePeriodValidator.new
+      doc_start = year_validator.measure_period_start(doc).value
+      unless doc_start
+        failed_files[name] = ['Document needs to report the Measurement Start Date']
+        return nil, 0
+        # doc_end = validator.measure_period_end(doc).value -> should we validate end???
       end
 
       # import
       patient = QRDA::Cat1::PatientImporter.instance.parse_cat1(doc)
-      patient.update(_type: CQM::VendorPatient, correlation_id: params[:vendor_id])
-      patient
+      patient.update(_type: CQM::VendorPatient, correlation_id: params[:vendor_id], bundleId: @bundle.id)
+
+      # shift date
+      utc_start = DateTime.parse(doc_start).to_time.utc
+      bundle_utc_start = DateTime.strptime(@bundle.measure_period_start.to_s, '%s').utc
+      time_dif = 0
+      # Compare date alone, without time
+      if utc_start.strftime('%x') != bundle_utc_start.strftime('%x')
+        patient.qdmPatient.shift_dates(time_dif)
+        time_dif = @bundle.measure_period_start - utc_start.to_i
```

Comment:
shouldn't 'time_dif' happen before patient.qdmPatient.shift_dates(time_dif)

Reply:
That's correct- sorry, last minute rubocop change that I didn't check through properly